### PR TITLE
Enhance/v2 sgd regressor classifier

### PR DESCRIPTION
**Pull request by JasonShin**
Context:
The pull request aims to improve SGDClassifier and SGDRegressor by refactoring the code to use Tensorflow.js's Linear Algebra API. In addition, it implements a few new constructor options including

- random_state: random seeding value
- loss: configurable loss function, defaults to L2

### CODE REVIEW

**Comment by JasonShin** [comment]
I am still working on the configurable loss function, should be ready soon.

**Comment by JasonShin** [comment]
```
private addBias(X, bias = 1): number[][] {
    // TODO: Is there a TF way to achieve it?

```

Let's just make a separate ticket for this.
